### DISCUSSION TITLE
gt/gtExtras: cols_label with_tooltip from a dataframe/list

### DISCUSSION THREAD

**Discussion by vikasrawal** [rstudio/gt#1474] [Q&A]
I want to pick up labels and tooltips for column names from a data.frame (which has three columns: colnames, collabels and coltooltips).

In other words, in the following,

cols_label(
    var1=with_tooltip(label="Label 1",tooltip="Tooltip 2"),
    var2=with_tooltip(label="Label 2",tooltip="Tooltip 2"))

I want to pick up variables, labels and tooltips from a data frame. I can write a function that can create these strings but how do I use them as arguments for cols_label.

Thanks

**Comment by olivroy** [comment]
This is a good idea.

In the mean time, you can use plain old html to do it!

If you use the abbr html tag https://html.com/tags/abbr/#:~:text=The%20element%20is%20used,technologies%20do%20use%20this%20information. and wrap it in the special `html()` helper function, you will create a tooltip!


```r
mtcars |> 
  gt::gt() |> 
  cols_label(
    vs = html(
      "<abbr title = 'Tooltip'>  vs</abbr>"
    )
  )
```
![image](https://github.com/rstudio/gt/assets/52606734/3df0b57a-db23-4b4f-a10b-21baaae4c276)
